includes/Database_Upgrader.php

```diff
@@ -122,309 +137,20 @@ public static function get_registration_action_priority() {
 	 *
 	 * @since 1.0.0
 	 *
-	 * @param string $routine         The method to call.
+	 * @param string $class           The Class to call.
 	 * @param string $version         The new version.
 	 * @param string $current_version The current set version.
 	 *
 	 * @return void
 	 */
-	protected function run_upgrade_routine( $routine, $version, $current_version ) {
+	protected function run_upgrade_routine( $class, $version, $current_version ) {
 		if ( version_compare( $current_version, $version, '<' ) ) {
-			$this->$routine( $current_version );
-		}
-	}
-
-	/**
-	 * First database migration.
-	 *
-	 * @since 1.0.0
-	 *
-	 * @return void
-	 */
-	protected function upgrade_1() {
-		// Do nothing.
-	}
-
-	/**
-	 * Replaces conic color type with linear.
-	 *
-	 * @since 1.0.0
-	 *
-	 * @return void
-	 */
-	protected function v_2_replace_conic_style_presets() {
-		$style_presets = get_option( Story_Post_Type::STYLE_PRESETS_OPTION, false );
-		// Nothing to do if style presets don't exist.
-		if ( ! $style_presets || ! is_array( $style_presets ) ) {
-			return;
-		}
-
-		$fill_colors = [];
-		$text_styles = [];
-		if ( ! empty( $style_presets['fillColors'] ) ) {
-			foreach ( $style_presets['fillColors'] as $color ) {
-				if ( ! isset( $color['type'] ) || 'conic' !== $color['type'] ) {
-					$text_styles[] = $color;
-					continue;
-				}
-				$updated_preset         = $color;
-				$updated_preset['type'] = 'linear';
-				$fill_colors[]          = $updated_preset;
-			}
-		}
-
-		if ( ! empty( $style_presets['textStyles'] ) ) {
-			foreach ( $style_presets['textStyles'] as $preset ) {
-				if ( empty( $preset['backgroundColor'] ) ) {
-					$text_styles[] = $preset;
-					continue;
-				}
-				$bg_color = $preset['backgroundColor'];
-				if ( ! isset( $bg_color['type'] ) || 'conic' !== $bg_color['type'] ) {
-					$text_styles[] = $preset;
-					continue;
-				}
-				$updated_preset                            = $preset;
-				$updated_preset['backgroundColor']['type'] = 'linear';
-				$text_styles[]                             = $updated_preset;
+			if ( ! method_exists( $this->injector, 'make' ) ) {
```

Comment:
Could move this condition to `register()` above I presume?

Reply:
Alternatively, is there a way to avoid the injector class property? Feel like it's not super needed.

Reply:
I originally tried this but sadly phpstan didn't like this.